apps/deploy-web/src/components/alerts/ContactPointsListContainer/ContactPointsListContainer.tsx

```diff
@@ -32,57 +30,60 @@ export const ContactPointsListContainer: FC<ContactPointsListContainerProps> = (
   const mutation = notificationsApi.v1.deleteContactPoint.useMutation();
   const notificator = useNotificator();
 
-  const handleRemove = useCallback(() => {
-    setIsRemoving(true);
-    mutation.mutate({
-      path: {
-        id: contactPointIdToRemove
+  const remove = useCallback(
+    async (id: ContactPoint["id"]) => {
+      try {
+        setIdsBeingRemoved(prev => new Set(prev).add(id));
+
+        await mutation.mutateAsync({
+          path: {
+            id
+          }
+        });
+
+        notificator.success("Contact point removed", { dataTestId: "contact-point-remove-success-notification" });
+
+        if (data?.data.length === 1 && page > 1) {
+          setPage(page - 1);
+        } else {
+          refetch();
+        }
+      } catch (error) {
+        notificator.error("Failed to remove contact point", {
+          dataTestId: "contact-point-remove-error-notification"
+        });
+      } finally {
+        setIdsBeingRemoved(prev => {
+          const nextSet = new Set(prev);
+          nextSet.delete(id);
+          return nextSet;
+        });
       }
-    });
-  }, [mutation, contactPointIdToRemove]);
+    },
+    [mutation, data?.data.length, page, refetch, notificator]
+  );
 
-  const handlePageChange = useCallback((page: number, pageSize: number) => {
+  const changePage = useCallback((page: number, pageSize: number) => {
     setPage(page + 1);
     setLimit(pageSize);
   }, []);
 
-  useWhen(mutation.isError, () => {
-    notificator.error("Failed to remove contact point", { dataTestId: "contact-point-remove-error-notification" });
-    setIsRemoving(false);
-  });
-
-  useWhen(mutation.isSuccess, () => {
-    notificator.success("Contact point removed", { dataTestId: "contact-point-remove-success-notification" });
-
-    if (data?.data.length === 1 && page > 1) {
-      setPage(page - 1);
-    } else {
-      refetch();
-    }
-
-    setContactPointIdToRemove(null);
-    setIsRemoving(false);
-  });
-
   return (
     <>
       {children({
-        data: data?.data || [],
         pagination: {
           page,
           limit,
           total: data?.pagination.total ?? 0,
           totalPages: data?.pagination.totalPages ?? 0
         },
-        onPageChange: handlePageChange,
+        data: data?.data || [],
+        onPageChange: changePage,
+        onRemove: remove,
+        idsBeingRemoved,
```

Comment:
non-blocking: how would you like to shorten a bit

```suggestion
        removingIds,
```

Reply:
Renamed